source/server/version.h.in

```diff
@@ -5,6 +5,7 @@ namespace nidevice_grpc {
 
 static constexpr const char* kNiDeviceGrpcFileVersion = "@PROJECT_VERSION_MAJOR@.@PROJECT_VERSION_MINOR@.@PROJECT_VERSION_PATCH@.0";
 static constexpr const char* kNiDeviceGrpcOriginalFileName = "ni_grpc_device_server.exe";
+static constexpr const char* kNiDeviceGrpcBranchName = "@GIT_BRANCH@";
```

Comment:
I would've expected to see the updated `generated/version.h` file show up in the changed files. Did you build locally before pushing up these changes?

Reply:
Oh, I see. You removed it in a separate commit. The reason I bring it up is because when the build gets past that other [Bug](https://dev.azure.com/ni/DevCentral/_workitems/edit/2640696) I think it will fail when we check that what's in source matches what's produced after the build: https://github.com/ni/grpc-device/blob/cbf3f281529daa68036e8f27863358697199f09b/.github/workflows/build_cmake.yml#L96

That's intended to make sure there's no non-deterministic codegen issues and that our what we're looking at in git for the source/generated folder is up to date.

This one feels a little different than the rest, though, because it would pass on your PR build and then fail on main. You'll probably need to relax that check to something like `git diff ':!generated/version.h' --exit-code`.

Reply:
I did, but removed it as I believed it would be generated with the PR/CI build. Should I locally build again and leave the version.h (with the new variable = users/christag-ni/noteDevWithCommand) to be pushed in this PR?

Reply:
Yeah, I realized what happened after my original comment. @astarche, do you think it makes sense to fix the build to not diff the generated/version.h file like I mentioned or just add generated/version.h to `.gitignore`? I'm kind of leaning towards just adding it to `.gitignore` so we don't see it pop up in every PR where they build locally.

Reply:
`.gitignore` definitely sounds correct to me.

Normally I think it would be helpful to put the git ignored files in a meaningfully named folder. But, in this case, the obvious options (like `.generated`? `generated/.volatile`? `generated/,version`?)  seem to raise more questions than answers or are redundant. The idea that "versions are special" is more intuitive than trying to create some other "special meaning".

So I'm happy with whatever makes sense to you. Or just leave `version.h` where it is if you don't have a better option.

Reply:
I've added `/generated/version.h` to .gitignore but it's still diffing. Do file paths change during PR builds?

Reply:
Ah, I didn't know this. This [SO post](https://stackoverflow.com/questions/17820056/git-diff-doesnt-ignore-specified-files-in-gitignore) summarizes it. You can do this locally, against your branch if you hand-edit `/generated/version.h` and then run `git diff` it will tell you it changed even though it's in `.gitignore` now.

So I think the order of operations needs to be:
1. Revert the commit to `.gitignore`
2. Remove the generated/version.h file -> `git rm -f generated/version.h` and commit that
3. Add back `.gitignore` change and commit that

I think that will properly remove the version.h file from being tracked in git (avoid it still popping up in git diff) and also prevent it from coming back in the future.